shell/common/surface.cc

```diff
@@ -51,6 +51,27 @@ bool SurfaceFrame::PerformSubmit() {
   return false;
 }
 
+Surface::Surface() : scale_(1.0) {}
+
 Surface::~Surface() = default;
 
+bool Surface::SupportsScaling() const {
+  return false;
+}
+
+double Surface::GetScale() const {
+  return scale_;
+}
+
+void Surface::SetScale(double scale) {
+  static const double kMaxScale = 1.0;
+  static const double kMinScale = 0.25;
```

Comment:
Should these be `constexpr`? IIRC the spec claims these satisfy the criteria for constant expressions anyway though.